test/test_choice.py

```diff
@@ -48,33 +58,34 @@ class BB(Base):
 def test_choice_with_dict():
     @dataclass
     class C(TestSetup):
-        option: Union[AA, BB] = choice({
+        option: Union[AA, BB, float] = choice({
             "a": AA("aaa"),
             "b": BB("bbb"),
             "bob": AA("bobobo"),
+            "f": 1.23
         }, default="a")
-    c = C.setup("--option a")
-    assert c.option.value == "aaa"
+        options: List[Union[AA, BB, float]] = choice({
+            "a": AA("aaa"),
+            "b": BB("bbb"),
+            "bob": AA("bobobo"),
+            "f": 1.23
+        }, default_factory=["a"].copy)
 
-    c = C.setup("--option bob")
-    assert c.option.value == "bobobo"
+    c = C.setup("")
+    assert c.option == AA("aaa")
+    assert c.options == [AA("aaa")]
 
-def test_choice_with_dict_weird():
-    @dataclass
-    class D(TestSetup):
-        option: List[Base] = choice({
```

Comment:
@lebrice I think this test was incorrect.
Type of `option` must be `Union[List[Base], float, Base]` because we specify here type of final attribute.

I've merged this test with `test_choice_with_dict` above.
If you disagree, I can fix it as you say

Reply:
Sure, this is fine. Thanks!